showkase-processor/src/main/java/com/airbnb/android/showkase/processor/ShowkaseProcessor.kt

```diff
@@ -121,6 +184,97 @@ class ShowkaseProcessor @JvmOverloads constructor(
             }.flatten().mapNotNull { it }.toSet()
     }
 
+    // The reason for this method to take both an annotation and to check for supported types
+    // Is that we want to check if the custom annotation is used in the same module
+    // as it was discovered.
+    @Suppress("ReturnCount")
+    private fun processCustomAnnotation(
+        roundEnvironment: XRoundEnv,
+        annotation: XTypeElement? = null
+    ): Set<ShowkaseMetadata.Component> {
+        val supportedTypes = mutableListOf<String>()
+        if (annotation != null) supportedTypes.add(annotation.qualifiedName)
+        supportedTypes.addAll(getSupportedMultipreviewTypes())
+
+        return supportedTypes.map { supportedAnnotation ->
+            val elementsAnnotated = roundEnvironment.getElementsAnnotatedWith(supportedAnnotation)
+            return@map elementsAnnotated.map elementScope@{ element ->
+                if (showkaseValidator.checkElementIsAnnotationClass(element)) {
+                    // Here we write to metadata to aggregate custom annotation data.
+                    // In this case, it would be a custom annotation that is annotated
+                    // with something custom. Eg. CombinedPreview.
+                    ShowkaseBrowserWriter(environment).writeCustomAnnotationElementToMetadata(
```

Comment:
We are writing in two places. Need to double check if we need to do this or doing it just once is sufficient

Reply:
Removed writing to file here. Seems like you where correct. We do not need it here :) Thanks for pointing that out, great catch! :)